app/fileSender.js

```diff
@@ -18,13 +18,11 @@ async function sendPassword(file, authKey, rawAuth) {
     const xhr = new XMLHttpRequest();
     xhr.onreadystatechange = () => {
       if (xhr.readyState === XMLHttpRequest.DONE) {
+        const nonce = xhr.getResponseHeader('WWW-Authenticate').split(' ')[1];
```

Comment:
There are cases where this header is not set (status == 404 for example), so would throw an exception here